dhcpv4/dhcpv4_test.go

```diff
@@ -355,6 +355,16 @@ func TestNewInform(t *testing.T) {
 	require.True(t, m.ClientIPAddr.Equal(localIP))
 }
 
+func TestDHCPv4NewInformWithModifier(t *testing.T) {
```

Comment:
Any reason to add this new test if nothing changed in this code?

Reply:
The use case in https://github.com/siderolabs/talos/pull/5897 (and the example in the updated description) requires supplying modifiers to `NewInform`, which has to date not been test-validated use case.

Reply:
Nothing changed in the `NewInform`. Do we really need a new test?

Reply:
As I'm improving `NewInform` here, although I'm not changing its functionality (to maintain backwards compatibility), as a consumer of the function I would like to have strong guarantees that it functions properly in all advertised cases. Even though it is empirically known to work properly right now, the test guarantees that modifier support won't be accidentally broken in the future by either a direct or indirect change. I do also not think that `TestDHCPv4NewInformWithModifier` would measurably increase test runtime as it is very lightweight.

Reply:
Is this strictly related to the title of this PR?

Reply:
I've updated the title of this PR.

Reply:
It would make more sense to me to have this in a separate PR instead of having such a long PR title :)

Reply:
Right, I can do that as well if you prefer, I was just thinking that that causes a lot of very small PRs which might cause a lot of spam and be hard to keep track of. Regardless, I'll try to find the time in the next days to split this PR up so that every change that can be described in a short PR title gets its own PR.

Reply:
Separated: https://github.com/insomniacslk/dhcp/pull/480